src/Imgix/UrlBuilder.cs

```diff
@@ -67,88 +73,189 @@ public String BuildUrl(String path, Dictionary<String, String> parameters)
             return GenerateUrl(Domain, path, parameters);
         }
 
+        private String GenerateUrl(String domain, String path, Dictionary<String, String> parameters)
+        {
+            String scheme = UseHttps ? "https" : "http";
+            path = path.TrimEnd('/').TrimStart('/');
+
+            var qs = GenerateUrlStringFromDict(parameters);
+            var localParams = new Dictionary<String, String>(parameters);
+
+            if (!String.IsNullOrEmpty(_signKey))
+            {
+                var hashString = String.Format("{0}/{1}{2}", _signKey, path, localParams.Any() ? "?" + qs : String.Empty);
+                localParams.Add("s", HashString(hashString));
+            }
+
+            return String.Format("{0}://{1}/{2}{3}", scheme, domain, path, localParams.Any() ? "?" + GenerateUrlStringFromDict(localParams) : String.Empty);
+        }
+
         public String BuildSrcSet(String path)
         {
             return BuildSrcSet(path, new Dictionary<string, string>());
         }
 
+        public String BuildSrcSet(String path, Dictionary<String, String> parameters)
+        {
+            return BuildSrcSet(path, parameters, MinWidth, MaxWidth, SrcSetWidthTolerance);
+        }
 
+        /// <summary>
+        /// Create a srcset attribute by disabling variable output quality.
+        ///
+        /// Variable output quality for dpr-based srcset attributes
+        /// is _on by default_. Setting `disableVariableQuality` to
+        /// `true` disables this.
+        /// 
+        /// </summary>
+        /// <param name="path">path to the image, i.e. "image/file.png"</param>
+        /// <param name="parameters">dictionary of query parameters</param>
+        /// <param name="disableVariableQuality">toggle variable quality output on
+        /// (default/false) or off (true).</param>
+        /// <returns></returns>
```

Comment:
Should this `<returns>` tag have a value?

Reply:
Yes!